malduck/extractor/config_builder.py

```diff
@@ -0,0 +1,538 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from maco.model import CategoryEnum, ConnUsageEnum, ExtractorModel
+
+if TYPE_CHECKING:
+    from .extractor import Extractor
+
+
+class ConfigBuilder:
+    """
+    Allows to build configuration imperatively, using
+    Maco model https://github.com/CybercentreCanada/Maco
+
+    .. versionadded:: 4.4.0
+    """
+
+    def __init__(self, parent: "Extractor") -> None:
+        self.parent = parent
+
+    def push_config(self, config: Dict[str, Any], exclude_family=True) -> None:
+        """
+        Pushes Maco-compatible configuration part to the extraction context.
+
+        :param config: Configuration part
+        :param exclude_family: |
+            By default 'family' is excluded because family match
+            is determined by extraction method (e.g. if it's decorated with
+            ``@weak``, it doesn't indicate a match).
+            If you actually want to indicate a match and ignore that decorator,
+            set exclude_family to False.
+        """
+        # 'family' is required field in ExtractorModel
+        # In our case 'family' is indicator that family has been actually matched
+        # which is determined by returned value from extractor method
+        config = ExtractorModel.parse_obj(
+            {**config, "family": self.parent.family}
+        ).dict(
+            exclude_defaults=True,
+            exclude={
+                "family",
+            }
+            if exclude_family
+            else set(),
+        )
+        self.parent.push_config(config, jsonable=False)
+
+    def add_other(self, others: Dict[str, Any]) -> None:
+        """
+        Add new fields to ``others`` key that doesn't match to any of
+        predefined categories.
+
+        ``others`` is merged similarly to configuration itself:
+
+        - list elements are treated like ordered sets and
+          are collecting all elements within extraction
+        - other elements can be set only once, override will
+          throw an exception
+
+        :param others: Dictionary with other configuration fields
+        """
+        return self.push_config(dict(others=others))
```

Comment:
Why do we add a special magic key called `others`? Would this work if we just did this instead:

```suggestion
        return self.push_config(others)
```

(of course there are some edge cases, like `add_other({"tcp": 3})`, but maybe it's a good tradeoff vs magic `others` key).

Reply:
Actually it's support for `other` collection that is meant to contain anything extra, not expected in the scheme: https://github.com/CybercentreCanada/Maco/blob/master/maco/model/model.py#L200.

And yeah, I see my mistake: the key is `other`, not `others`